### PR TITLE
Connect email form to server

### PR DESCRIPTION
**Pull request by nathan-barrett**
This PR connects the email form to https://www.mozilla.org/en-US/email-mieco/ (thanks @robhudson!). I've mainly used the newsletter scripts from bedrock.

On a successful POST request, you should get a `200` back

### CODE REVIEW

File: scripts/form-utils.js [hunk 1]
```diff
@@ -101,36 +103,21 @@ function postToBasket(email, params, url, successCallback, errorCallback) {
 
     xhr.onerror = errorCallback;
     xhr.open('POST', url, true);
-    xhr.setRequestHeader('Content-type', 'application/x-www-form-urlencoded');
+    xhr.setRequestHeader('Content-type', 'application/json');
     xhr.setRequestHeader('X-Requested-With', 'XMLHttpRequest');
+    xhr.setRequestHeader("Access-Control-Allow-Headers", "*");
```
Comment (alexgibson) [inline scripts/form-utils.js]: Suggestion: I think you can remove this here (it's bedrock that needs to set it I believe?)

File: scripts/form-utils.js [hunk 2]
```diff
@@ -55,15 +55,17 @@ function disableFormFields(form) {
...
 
     for (let i = 0; i < formFields.length; i++) {
         formFields[i].disabled = false;
     }
 }
 
-function postToBasket(email, params, url, successCallback, errorCallback) {
+function postToEmailServer(params, successCallback, errorCallback) {
     const xhr = new XMLHttpRequest();
+    const url = "https://www-demo2.allizom.org/email-meico/";
```
Comment (alexgibson) [inline scripts/form-utils.js]: Suggestion: include locale (which will always be `en-US` for this view?) to make request a bit faster

```suggestion
    const url = "https://www-demo2.allizom.org/en-US/email-meico/";
```

**Review by alexgibson — COMMENTED** [review]
I took a look at this and I think it's bedrock that might need to set some extra headers possibly, when returning the pre-flight request:

`A cross-origin resource sharing (CORS) request was blocked because it contained request headers that were neither CORS-safelisted (Accept, Accept-Language, Content-Language, Content-Type) nor allowed by the Access-Control-Allow-Headers response header of the associated preflight request.
To fix this issue, include the additional request headers you want to use in the Access-Control-Allow-Headers response header of the associated preflight request.`

File: scripts/form-utils.js (hunk 1 above)
Comment (robhudson) [reply scripts/form-utils.js]: Yep, I'm not sure that makes sense for the request. I have it in the response from bedrock. I also added the `Access-Control-Allow-Headers` list and pushing to demo2 for more testing in a bit.

File: scripts/form-utils.js (hunk 2 above)
Comment (nathan-barrett) [reply scripts/form-utils.js]: Good idea!

**Review by alexgibson — COMMENTED** [review]
It looks like this needs a rebase

I tried testing the form, but until the bedrock changes are in production I can't get a successful 200 POST to verify things are working OK. I tried setting the POST destination to www-demo2, but for some reason I get a 400 error there when trying to submit the form. So I can't be confident this works all OK. 

At a high-level, the code looks good.

**Comment by robhudson** [comment]
The demo server will get a 400 since the mail server isn't set up there. If you get a 400 "Connection error", then all other checks have passed and that's a connection error to the mail server, which is to be expected. At that point, we attempted to send the email but it failed. We can verify the email as we roll out towards production.

File: scripts/email.js [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+        // Confirm privacy policy is checked
+        if (!privacy) {
+        EmailForm.handleFormError('Privacy policy not checked');
+            return false;
+        }
+
+        return true;
+    },
+
+    sumbit: (e) => {
```
Comment (reemhamz) [inline scripts/email.js:67]: I may have found the reason it wasn't POST-ing and not giving us a 200 status code.
You misspelled "submit"!
```suggestion
    submit: (e) => {
```

Comment (alexgibson) [inline scripts/email.js]: The function is misspelt both here and where the event is bound, so whilst the spelling should be corrected, I’m not sure that’s the reason.

Comment (nathan-barrett) [reply scripts/email.js]: 🙃

Comment (reemhamz) [reply scripts/email.js]: Nevermind, it's deeper than that :)

**Review by reemhamz — COMMENTED** [review]
Frontend looks good to me, aside from the email stuff.
r+! 🦃

**Review by reemhamz — APPROVED** [review]

**Pull request merged**